servers/audio/audio_server_sw.cpp

```diff
@@ -726,7 +726,8 @@ void AudioServerSW::_thread_func(void *self) {
 
 	while (!as->exit_update_thread) {
 		as->_update_streams(true);
-		OS::get_singleton()->delay_usec(5000);
+		if (OS::get_singleton())
+			OS::get_singleton()->delay_usec(5000);
```

Comment:
This looks a little suspect. We just sleep in hopes of avoiding a race?

Reply:
It sleeps a little to avoid calling _update_streams all the time and eat a lot of cpu.

Reply:
I could change this if (OS::get_singleton()) for an ERR_BREAK.

Reply:
If it isn't to avoid a race I guess this is OK, but how do we know this is the right amount of time to wait? Can't we have some kind of callback?

Reply:
We process the audio in its own thread, which I think its great we take advantage of multiple cores and also avoid taking process time from the main thread, so a callback would be a bad idea in my opinion. And the timer has been like this for a long time I think so I don't want to mess with it, and 5ms if fine in my opinion.